docs/use_cases/embeddings_on_browser.md

```diff
@@ -0,0 +1,494 @@
+            ctx.fillRect(j * cellSize, i * cellSize, cellSize, cellSize);
+          }
+        }
+      }
+    }
+  }, [similarityMatrix, canvasSize]);
+```
+
+### User Input Section
+
+This code represents a part of the user interface where users can input multiple sentences. It includes a label, a multiline text input field, and the ability to control and update the input through React state management. The user's entered sentences are stored in the **`sentences`** state variable and can be used for further processing in the component.
+
+```tsx
+{/* User Input Section */}
+      <Grid item md={6}>
+
+        // Heading
+        <Typography variant="h2" gutterBottom>Encode Sentences</Typography>
+
+        // Multiline text input
+        <TextField
+          id="sentencesInput"
+          label="Multiline"
+          multiline
+          fullWidth
+          rows={6}
+          value={sentences}
+          onChange={(e) => setSentences(e.target.value)}
+        />
+      </Grid>
+```
+
+### Embeddings Output Section
+
+The UI embeddings output section displays the embeddings stored in the **`embeddings`** state variable, including a label, a multiline text output field. Through the embeddings output section you can control and update the displayed content using React state management.
+
+
+```tsx
+      {/* Embeddings Output Section */}
+      <Grid item md={6}>
+
+        // Heading
+        <Typography variant="h2" gutterBottom>Embeddings</Typography>
+
+        // Multiline text field to display embeddings
+        <TextField
+          id="embeddingOutput"
+          label="Multiline"
+          multiline
+          fullWidth
+          rows={6}
+          value={embeddings}
+          variant="filled"
+        />
+      </Grid>
+```
+
+### Generate Embedding Button
+
+The following code represents a raised, solid button in the UI that triggers the **`handleGenerateEmbedding`** function to initiate the embedding generation process. The generate embedding button is initially disabled if there are no input sentences (**`!sentences`**) or if the model is currently loading (**`modelLoading`**).
+
+```tsx
+      {/* Generate Embedding Button */}
+      <Grid item xs={6}>
+        <Button
+          variant="contained"
+          id='generate-embedding'
+          onClick={handleGenerateEmbedding}
+          disabled={!sentences || modelLoading}
+        >
+          Generate Embedding
+        </Button>
+      </Grid>
+```
+
+### Model Indicator
+
+This code controls what is displayed in the user interface based on the values of the **`modelComputing`** and **`modelLoading`** state variables. If  is **`true`**, it first checks if . If it is, a loading indicator is displayed. If **`false`**, a message indicating that the model is loaded is shown. If , nothing is rendered in this section. This conditional rendering allows the user to see either a loading indicator or a model loaded message based on the status of model loading and computing.
```

Comment:
line 392
1. ” If  is **`true`**, it first checks if . ”
This is in your original commit as well and never fixed. If what is true, it first checks if what ?

2. ” If , nothing is rendered in this section. ”
If what?